triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/TriggerDao.scala

```diff
@@ -54,7 +54,28 @@ object TriggerDao {
           package bytea not null
         )
       """
+    val createPartyIndex: Fragment = sql"""
+        create index triggers_by_party on running_triggers(party_token)
+      """
     (createTriggerTable.update.run
-      *> createDalfTable.update.run).void
+      *> createDalfTable.update.run
+      *> createPartyIndex.update.run).void
+  }
+
+  def addRunningTrigger(t: RunningTrigger): ConnectionIO[Unit] = {
+    val partyToken = t.jwt.value
+    val fullTriggerName = t.triggerName.toString
+    val insertTrigger: Fragment = Fragment.const(
+      s"insert into running_triggers values ('${t.triggerInstance}', '$partyToken', '$fullTriggerName')"
+    )
+    insertTrigger.update.run.void
+  }
+
+  def getTriggersForParty(partyToken: Jwt): ConnectionIO[Vector[UUID]] = {
+    val token = partyToken.value
+    val list: Fragment = Fragment.const(
+      s"select trigger_instance from running_triggers where party_token = '$token'"
```

Comment:
You probably want to order this by something. Unordered selects often come back to bite you later.

Reply:
Ok good point. I'll `order by trigger_instance`.